File: src/main/java/bad/robot/radiate/monitor/Monitor.java
package bad.robot.radiate.monitor;

import java.util.List;
import java.util.concurrent.ScheduledFuture;

@Deprecated
public interface Monitor {
    Iterable<ScheduledFuture<?>> start(Iterable<MonitoringTask> tasks);
    void cancel(Iterable<ScheduledFuture<?>> tasks);
    List<Runnable> stop();
}

File: src/main/scala/bad/robot/radiate/monitor/Monitor.scala
package bad.robot.radiate.monitor

import java.util.concurrent.ScheduledFuture

trait MonitorS {
  def start(tasks: List[MonitoringTaskS]): List[ScheduledFuture[_]]
  def cancel(tasks: List[ScheduledFuture[_]])
  def stop: List[Runnable]
}
